x-pack/plugins/security/public/management/roles/edit_role/validate_role.test.ts

```diff
@@ -98,6 +98,85 @@ describe('validateRoleName', () => {
   });
 });
 
+describe('validateRoleName for serverless', () => {
+  beforeEach(() => {
+    validator = new RoleValidator({ shouldValidate: true, buildFlavor: 'serverless' });
+  });
+  test('should not allow whitespace', () => {
```

Comment:
Nit: for full test coverage, I would expect to test leading and trailing whitespace as well.